src/System.Private.Reflection.Core/src/System/Reflection/Runtime/Assemblies/EcmaFormat/EcmaFormatRuntimeAssembly.cs

```diff
@@ -33,10 +33,13 @@ namespace System.Reflection.Runtime.Assemblies.EcmaFormat
 {
     internal sealed partial class EcmaFormatRuntimeAssembly : RuntimeAssembly
     {
-        private EcmaFormatRuntimeAssembly(MetadataReader reader)
+        public override string Location { get; }
+
+        private EcmaFormatRuntimeAssembly(MetadataReader reader, string assemblyPath)
         {
             AssemblyDefinition = reader.GetAssemblyDefinition();
             MetadataReader = reader;
+            Location = assemblyPath ?? "";
```

Comment:
Nit: It may look better to have `_assemblyPath` field that is null when no path is specified. And normalize it to empty string in the `string Location` property implementation.